src/viewportchecker.ts

```diff
@@ -180,8 +180,17 @@ export default class ViewportChecker implements EventListenerObject {
             // options.
             for (const opt in attrOptionMap) {
                 const dataKey = attrOptionMap[opt as keyof typeof attrOptionMap];
-                const val = $obj.dataset[dataKey];
-                if (val) { (objOptions as any)[opt] = val; };
+
+                const val = 
+                    typeof $obj.dataset[dataKey] !== undefined ? 
+                    (
+                        $obj.dataset[dataKey] === "true" || $obj.dataset[dataKey] === "false" ? 
+                        getBooleanStringRepresentation($obj.dataset[dataKey]!) : $obj.dataset[dataKey]
+                    ) : undefined;
```

Comment:
For this one some of the checks can be removed, since they're implicitly also being done by your `true/false` expression:

```typescript
const val = ["true", "false"].includes($obj.dataset[dataKey])
   ? getBooleanStringRepresentation($obj.dataset[dataKey]!)
   :  $obj.dataset[dataKey];
```

This will check if the value is `true` or `false` and if so pass it on to your coercion function. If not it will just pass along the original value (which could be undefined or any other value)